test/e2e/scenario/recorder/shadowDom.scenario.ts

```diff
@@ -0,0 +1,232 @@
+        expect(insideTextContent).toBe('toto')
+      })
+  })
+  describe('incremental snapshot', () => {
+    createTest('record click')
+      .withRum({ enableExperimentalFeatures: ['recordShadowDom'] })
+      .withRumInit(initRumAndStartRecording)
+      .withSetup(bundleSetup)
+      .withBody(
+        html`
+          ${simpleShadowDom}
+          <my-open-web-component id="some" />
+        `
+      )
+      .run(async ({ serverEvents }) => {
+        await browserExecute(() => document.documentElement.outerHTML)
+        const label = await getNodeInsideShadowDom('label')
+        await label.click()
+        await flushEvents()
+        expect(serverEvents.sessionReplay.length).toBe(1)
+        const fullSnapshot = findFullSnapshot(getFirstSegment(serverEvents))!
+        const labelNode = findElementWithTagName(fullSnapshot.data.node, 'label')!
+        const mouseInteraction = findIncrementalSnapshot(
+          getFirstSegment(serverEvents),
+          IncrementalSource.MouseInteraction
+        )!
+        expect(mouseInteraction).toBeTruthy()
+        expect(mouseInteraction.data.source).toBe(IncrementalSource.MouseInteraction)
+        expect((mouseInteraction.data as MouseInteractionData).id).toBe(labelNode.id)
+      })
+
+    createTest('record input')
+      .withRum({ defaultPrivacyLevel: 'allow', enableExperimentalFeatures: ['recordShadowDom'] })
+      .withRumInit(initRumAndStartRecording)
+      .withSetup(bundleSetup)
+      .withBody(
+        html`
+          ${simpleShadowDom}
+          <my-open-web-component id="some" />
+        `
+      )
+      .run(async ({ serverEvents }) => {
+        await browserExecute(() => document.documentElement.outerHTML)
+        const input = await getNodeInsideShadowDom('input')
+        await input.addValue('t')
+
+        await flushEvents()
+        expect(serverEvents.sessionReplay.length).toBe(1)
+        const fullSnapshot = findFullSnapshot(getFirstSegment(serverEvents))!
+        const inputNode = findElementWithTagName(fullSnapshot.data.node, 'input')!
+        const inputRecord = findIncrementalSnapshot(getFirstSegment(serverEvents), IncrementalSource.Input)!
+        expect(inputRecord).toBeTruthy()
+        expect(inputRecord.data.source).toBe(IncrementalSource.Input)
+        expect((inputRecord.data as InputData).id).toBe(inputNode.id)
+        expect((inputRecord.data as { text: string }).text).toBe('totot')
+      })
+
+    createTest('record mutation')
+      .withRum({ defaultPrivacyLevel: 'allow', enableExperimentalFeatures: ['recordShadowDom'] })
+      .withRumInit(initRumAndStartRecording)
+      .withSetup(bundleSetup)
+      .withBody(
+        html`
+          ${simpleShadowDom}
+          <my-open-web-component id="some" />
+        `
+      )
+      .run(async ({ serverEvents }) => {
+        await browserExecute(() => document.documentElement.outerHTML)
+        const input = await getNodeInsideShadowDom('input')
+        await input.addValue('t')
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏make a mutation without triggering other events, so the test case stays simpler/focused.